operator/v1/stable.console.testsuite.yaml

```diff
@@ -14,3 +14,114 @@ tests:
+      kind: Console
+      spec:
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enabled
+          - id: admin
+            visibility: 
+              state: Disabled
+    expected: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        logLevel: Normal
+        operatorLogLevel: Normal
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enabled
+          - id: admin
+            visibility: 
+              state: Disabled
+  - name: Should throw an error for incorrect value of state in perspectives
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enables
+    expectedError: "spec.customization.perspectives[0].visibility.state: Unsupported value: \"Enables\": supported values: \"Enabled\", \"Disabled\", \"AccessReview\""
+  - name: Should be able to add pinned resources to a perspective
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enabled
+            pinnedResources:
+            - group: ""
+              resource: configmaps
+              version: v1
+    expected: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        logLevel: Normal
+        operatorLogLevel: Normal
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enabled
+            pinnedResources:
+            - group: ""
+              resource: configmaps
+              version: v1
+  - name: Should not be able to add pinned resources to "admin" perspective
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        customization:
+          perspectives:
+          - id: admin
+            visibility: 
+              state: Enabled
+            pinnedResources:
+            - group: ""
+              resource: configmaps
+              version: v1
+    expectedError: "pinnedResources is forbidden when id is admin"
+  - name: Should throw an error if "group" is missing from the pinnedResources
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: Console
+      spec:
+        customization:
+          perspectives:
+          - id: dev
+            visibility: 
+              state: Enabled
+            pinnedResources:
+            - resource: configmaps
+              version: v1
+    expectedError: "spec.customization.perspectives[0].pinnedResources[0].group: Required value"
+  - name: Should throw an error if the value of "version" in the pinnedResources doesnot match the required regex
```

Comment:
You need validation of the regexes for the group and resource fields too please

Reply:
done